src/irc/core/irc-cap.c

```diff
@@ -0,0 +1,229 @@
+#include "module.h"
+#include "signals.h"
+#include "misc.h"
+
+#include "irc-cap.h"
+#include "irc-servers.h"
+
+int cap_toggle (IRC_SERVER_REC *server, char *cap, int enable) {
+	/* This is meant to be used after the inital negotiation is done */
+	if (!server->cap_complete)
+		return FALSE;
+
+	if (enable && g_slist_find(server->cap_supported, cap)) {
+		irc_send_cmdv(server, "CAP REQ %s", cap);
```

Comment:
Shouldn't this also add to the requested cap to `server->cap_active`? Since receiving the ACK doesn't do that atm